src/routes/tools/_common/helpers.js

```diff
@@ -234,3 +235,28 @@ export const convertAnnualRateToSum = ({ date, value }) => {
     return value * 365;
   }
 };
+
+/**
+ * Create initial configuration
+ * @param {object} defaultParams - default params for tool
+ * @param {object} urlParams - params generated from url query string by sapper
+ * @return {object}
+ */
+// Helper function to create an object with initial settings for a Cal-Adapt tool
+export const getInitialConfig = (
+  urlParams = {},
+  defaultParams = INITIAL_CONFIG
+) => {
+  if (Object.keys(urlParams).length === 0) {
+    return defaultParams;
+  }
+
+  const { models, months, ...rest } = urlParams;
+
+  return {
+    ...defaultParams,
+    ...rest,
+    ...(models && { models: models.split(",") }),
```

Comment:
@clhenrick I added validation functions for `models` and `months`. Let me know what you think.

```
function validateModels(value) {
  if (typeof value !== "string" || !value.split(",").length) {
    return false;
  }
  const models = PRIORITY_10_MODELS.map(({ id }) => id);
  return value.split(",").every((d) => models.includes(d));
}

function validateMonths(value) {
  if (typeof value !== "string" || !value.split(",").length) {
    return false;
  }
  const range = extent(MONTHS_LIST, (d) => d.id);
  return value.split(",").every((d) => +d >= range[0] && +d <= range[1]);
}
```

Reply:
That's definitely a more robust way of checking those properties. The only improvement I would try to make is to reduce the number of times `String.prototype.split` is used. Currently it will be used 3 times in each validation method. 

I might try doing the following:

1. test if it's a string and has a `length`
2. make a local variable for the array from `value.split(", ")`
3. do the validation test over each item in the array
4. if it's valid, add the property to the config object
5. if not then don't add the property to the config object

That way you would only need to use `.split(",")` once, but it might mean you have to pass the config object into the validation helper function instead of using the optional object property syntax with the spread operator.

There's probably another way to do it too, let me know if you think of it, no pressure to do any of this though either.